src/contentScript/nativeEditor/elementPicker.ts

```diff
@@ -48,164 +51,200 @@ interface UserSelection {
-        filteredElements = [...document.querySelectorAll<HTMLElement>(filter)];
-        const updateOverlay = () => {
-          if (!_cancelSelect) {
-            // The operation has completed
-            return;
+      function prehiglightItems() {
+        let filteredElements: HTMLElement[];
+        if (filter) {
+          filteredElements = [
+            ...document.querySelectorAll<HTMLElement>(filter),
+          ];
+          const updateOverlay = () => {
+            if (!_cancelSelect) {
+              // The operation has completed
+              return;
+            }
+
+            overlay.inspect(filteredElements);
+            setTimeout(() => requestAnimationFrame(updateOverlay), 30); // Only when the tab is visible
+          };
+
+          if (filteredElements.length > 0) {
+            updateOverlay();
           }
-
-          overlay.inspect(filteredElements);
-          setTimeout(() => requestAnimationFrame(updateOverlay), 30); // Only when the tab is visible
-        };
-
-        if (filteredElements.length > 0) {
-          updateOverlay();
         }
       }
-    }
 
-    function findExpectedTarget(target: EventTarget): HTMLElement | void {
-      if (!(target instanceof HTMLElement)) {
-        return;
-      }
+      function findExpectedTarget(target: EventTarget): HTMLElement | void {
+        if (!(target instanceof HTMLElement)) {
+          return;
+        }
 
-      if (!filter) {
-        return target;
-      }
+        if (!filter) {
+          return target;
+        }
 
-      return target.closest<HTMLElement>(filter);
-    }
+        return target.closest<HTMLElement>(filter);
+      }
 
-    function startInspectingNative() {
-      _cancelSelect = cancel;
-      registerListenersOnWindow(window);
-      addInspectingModeStyles(window);
-      prehiglightItems();
-    }
+      function startInspectingNative() {
+        _cancelSelect = cancel;
+        registerListenersOnWindow(window);
+        addInspectingModeStyles(window);
+        if (enableSelectionTools) {
+          addMultiSelectionTool(window);
+        }
 
-    function stopInspectingNative() {
-      hideOverlay();
-      _cancelSelect = null;
-      removeListenersOnWindow(window);
-      removeInspectingModeStyles();
-    }
+        prehiglightItems();
+      }
 
-    function onClick(event: MouseEvent) {
-      const target = findExpectedTarget(event.target);
-      if (event.altKey || !target) {
-        return;
+      function stopInspectingNative() {
+        hideOverlay();
+        _cancelSelect = null;
+        removeListenersOnWindow(window);
+        removeInspectingModeStyles();
+        if (enableSelectionTools) {
+          removeMultiSelectionTool();
+        }
       }
 
-      event.preventDefault();
-      event.stopPropagation();
+      function handleDone(target?: HTMLElement) {
+        try {
+          const result = target
+            ? uniq([...targets, target])
+            : uniq([...targets]);
```

Comment:
```suggestion
          const result = uniq(compact([...targets, target]))
```
I think this would work also if you want to simplify the ternary operator.